src/components/MapView/MapView.web.tsx

```diff
@@ -40,8 +40,17 @@ const MapView = forwardRef<MapViewHandle, MapViewProps>(
 
             const map = mapRef.getMap();
 
+            const resizeObserver = new ResizeObserver(() => {
+                mapRef?.resize();
+            });
+            resizeObserver.observe(mapRef?.getContainer());
```

Comment:
This looks great, but shouldn't we extract it to another useEffect call with only the mapRef as the dependency? We have some early returns above that might prevent setting up the resize observer; for example when there is one waypoint.

Reply:
@neil-marcellini i updated the code.kindly review it again pls.